torch/distributed/algorithms/model_averaging/utils.py

```diff
@@ -1,13 +1,12 @@
 # flake8: noqa C101
 import itertools
-from typing import Iterator
+from typing import List
 
 import torch
 import torch.distributed as dist
 
-
 def average_parameters(
-    params: Iterator[torch.nn.Parameter], process_group: dist.ProcessGroup
+    params: List[torch.nn.Parameter], process_group: dist.ProcessGroup
```

Comment:
Just want to double confirm: if the input is only an iterator, then you will see a regression in your use case, correct?

Reply:
the new function is following:
def average_parameters_or_parameter_groups(params: Union[Iterable[torch.nn.Parameter], Iterable[Dict[str, torch.nn.Parameter]]], process_group: dist.ProcessGroup):
    """
    average parameters or parameter_groups
    """
    filter_params = get_params_to_average(params)
    average_parameters(filter_params, process_group)

since filter_params is a list, so no need to iter filter_params.
so i change params type from Iterator[torch.nn.Parameter] to List[torch.nn.Parameter].
Of course the params argument can accept Iterator[torch.nn.Parameter] input.  
so what is your suggest?  keep Iterator or List ?